app/src/main/java/at/aau/serg/activities/GameScreenActivity.kt

```diff
@@ -93,34 +104,205 @@ class GameScreenActivity : AppCompatActivity() {
+    }
+    private fun moveCardPosition4To5(){
+        switchCards(R.id.ivPlayer4Card, R.id.ivPlayer5Card)
+        moveCardPosition3To4()
+    }
+
+    private fun moveCardPosition5To6(){
+        switchCards(R.id.ivPlayer5Card, R.id.ivPlayer6Card)
+        moveCardPosition4To5()
+    }
+
+
+    private fun switchCards(ivFromId: Int, ivToId: Int){
+        val ivFrom = findViewById<ImageView>(ivFromId)
+        val ivTo = findViewById<ImageView>(ivToId)
+        val resource = ivFrom.tag.toString().toInt()
+        runOnUiThread{
+            ivTo.setImageResource(resource)
+            ivTo.tag = ivFrom.tag
+        }
+    }
+
+    private fun setPlayerCard(cardImageView: ImageView, cardResourceId: Int){
+        runOnUiThread {
+            cardImageView.setImageResource(cardResourceId.takeIf { it != 0 } ?: R.drawable.card_diamonds_1)
+            cardImageView.tag = cardResourceId
+        }
     }
 
     private fun trickPrediction(socketResponse: Array<Any>) {
         Log.d("Socket", "Received trickPrediction event")
     }
+
+    private fun nextSubRound(socketResponse: Array<Any>){
+        Log.d("Socket", "Received nextSubRound event")
+
+        setPlayerGameScreen()
+
+        clearCardPlayedEvents()
+        isNextPlayer(socketResponse[0] as Int)
+
+    }
+
+    private fun isNextPlayer(nextPlayerIdx: Int){
+        if (nextPlayerIdx == myPlayerIndex) {
+            allowedToPlayCard = true
+            runOnUiThread {
+                Toast.makeText(this, "You are now!", Toast.LENGTH_LONG).show()
+            }
+        }
+    }
+
+    private fun startRound(socketResponse: Array<Any>) {
+        Log.d("Socket", "Received startRound event")
+        val gameData = (socketResponse[0] as JSONObject)
+        val cards = gameData.getJSONArray("hands").getJSONArray(myPlayerIndex)
+        val trumpCard = convertCard(gameData.getJSONObject("trump"))
+
+        if (cards != null) {
+            setCards(convertCards(cards))
+        }
+
+        val trumpImageView: ImageView = findViewById(R.id.ivTrumpCard)
+        val cardResourceId = trumpImageView.context.resources.getIdentifier("card_${trumpCard.suit.toString().lowercase()}_${trumpCard.value}", "drawable", trumpImageView.context.packageName)
+        setPlayerCard(trumpImageView, cardResourceId)
+
+        val newFragment = TrickPredictionFragment()
+        clearCardPlayedEvents()
+        updateFragmentContainerView(newFragment)
+    }
+
+    private fun nextPlayer(socketResponse: Array<Any>){
+        Log.d("Socket", "Received nextPlayer event ")
+        isNextPlayer(socketResponse[0] as Int)
+    }
+
+    private fun setCards(cards: Array<CardItem>){
+        runOnUiThread {
+            cardsViewModel.setCards(cards)
+        }
+    }
+
+    private fun clearCardPlayedEvents(){
+        cardPlayed = false
+        lastPlayedCard = null
+        countPlayedCards = 0
+    }
+
+    private fun convertCards(cardsJsonArray: JSONArray): Array<CardItem> {
```

Comment:
This is a copy of the same named function inside the `LobbyActivity`, please move it into a util class so we don't duplicate it.